userdata/userdata_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2023 Steffen Vogel.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package userdata_test
+
+import (
+	"bytes"
+	"encoding/hex"
+	"testing"
+
+	"github.com/google/nftables"
+	"github.com/google/nftables/userdata"
+)
+
+func TestUserDataComment(t *testing.T) {
+	r := nftables.Rule{}
+
+	wantComment := "this is my comment"
+	want := []byte{0, 19, 116, 104, 105, 115, 32, 105, 115, 32, 109, 121, 32, 99, 111, 109, 109, 101, 110, 116, 0}
```

Comment:
Can you construct the human-readable part of the `want` byte slice by using []byte(wantComment) so that it’s clearer to see please?

Reply:
Done.